bluetooth: hci: place tx/rx message buffer in .noinit section

ndrs-pst opened the pull request:
This change relocates the tx/rx message buffer from the `.bss` section to the `.noinit` section.
This adjustment is aimed at reducing boot time by lowering the size of the `.bss` section, without impacting the operational functionality of the buffer.

ndrs-pst committed:
bluetooth: hci: place tx/rx message buffer in .noinit section

This change relocates the tx/rx message buffer from the .bss section
to the .noinit section. This adjustment is aimed at reducing boot time by
lowering the size of the .bss section, without impacting the operational
functionality of the buffer.

Signed-off-by: Pisit Sawangvonganan <pisit@ndrsolution.com>

erwango commented:
^^@HoZel

jhedberg commented on drivers/bluetooth/hci/hci_ambiq.c:
```diff
@@ -52,7 +52,7 @@ LOG_MODULE_REGISTER(bt_hci_driver);
 #define SPI_MAX_RX_MSG_LEN 258
 
-static uint8_t g_hciRxMsg[SPI_MAX_RX_MSG_LEN];
+static uint8_t __noinit g_hciRxMsg[SPI_MAX_RX_MSG_LEN];
```
Not really related to this PR, but would be nice if someone could submit a patch to fix the naming of this variable, since it violates the Zephyr coding style. Using the same `rxmsg` name as `hci_spi_st.c` uses would be fine.

ndrs-pst replied:
@jhedberg, Is it OK to include in this PR? May separate as another commit.

jhedberg replied:
> @jhedberg, Is it OK to include in this PR? May separate as another commit.

Could be a second commit in this PR, or a new separate PR. I'm fine either way.

HoZHel commented:
LGTM, just you can also add these changes to `spi.c` driver.

ndrs-pst commented:
> LGTM, just you can also add these changes to `spi.c` driver.

OK, will include in this PR too.

ndrs-pst committed:
drivers: bluetooth: ambiq: fix the naming of rx message buffer

This change renames the RX message buffer from g_hciRxMsg to rxmsg,
as it does not follow the Zephyr coding style.

Signed-off-by: Pisit Sawangvonganan <pisit@ndrsolution.com>

carlescufi commented:
@jori-nordic PTAL
